Kconfig

```diff
@@ -0,0 +1,65 @@
+mainmenu "SOF $(PROJECTVERSION) Configuration"
+
+comment "Compiler: $(CC_VERSION_TEXT)"
+
+#source "scripts/Kconfig.include"
+
+source "src/Kconfig"
+
+config HOST_PTABLE
+	bool "Handling host page table"
+	default n
+	help
+	  Select for host page table
+
+config TASK_HAVE_PRIORITY_MEDIUM
+	bool "Handling medium priority task"
+	default n
+	help
+	  Select for medium priority task
+
+config BOOT_LOADER
+	bool "Boot Loader"
+	default n
+	help
+	  Select for boot loader
+
+config IRQ_MAP
+	bool "IRQ maps"
+	default n
+	help
+	  Select for IRQ maps
+
+config DMA_GW
+	bool "DMA Gateway"
+	default n
+	help
+	  Select for dma gateway
+
+config MEM_WND
+	bool "Memory Windows"
+	default n
+	help
+	  Select for memory windows
+
```

Comment:
I will make it like you wrote. It looks like these selects in Linux are used mostly for HAVE_* configs, because they declare capabilites of arch/platform, however I think in the future we should make Kconfig files less platform-oriented.

Problem is in how you define user. What you wrote is good for someone that want to run feature-customized version of FW on already defined platform, but for adding new platform with different HW capabilites it may become unmanagable.

Maybe we should aim for definitions based on multiple configs where you have base+diffs (something more like zephyr than linux).

Reply:
@jajanusz Might be easier to have a complete derived config for a platform/board, easier to read complete recipe. Base rather does not change, so no issue with multiple edits.

Another note: how about storing platform/board defs in /platforms, or /boards, not that deep under arch (what about heterogeneous plaforms)?

Reply:
@mmaka1 is right that storing it in arch may not be the best idea (even if it's how it's done in kernel)

@mmaka1 @lgirdwood We can always easily change things like these when PR is merged. Kconfig/Cmake are just tools, we can use them however we want for our project.

Reply:
Fwiw, Linux does store it's def_configs under arch/<name>/configs/ but I don't think we need to do this atm (as Linux easily has hundreds).
@jajanusz yeah, I'm keen to merge and sort out any integration issues after merging, that way all the others can try/test and provide feedback.
@mmaka1 do you approve ? if so I will merge.

Reply:
@jajanusz if you are also happy for merging now pls change the PR title to remove the DO NOT MERGE. Thanks.

Reply:
@lgirdwood 
Still waiting for person responsible for CI to update Internal CI.
I'll let you know when it's done.